dvc/repo/ls.py

```diff
@@ -85,7 +85,7 @@ def _ls(
     ret = {}
     for name, info in infos.items():
         dvc_info = info.get("dvc_info", {})
-        if dvc_info.get("entries") or not dvc_only:
+        if fs.isdvc(info["name"], recursive=True) or not dvc_only:
```

Comment:
Also a bit odd. Noted this in dvc-data, we'll solve the isdvc vs info["isdvc"] inconsistency there and then will get back here.